### PR TITLE
Create design doc for public flake rate system

### PR DESCRIPTION
**Pull request by andriyDev**
discusses design implemented for #11434.

### CODE REVIEW

File: enhancements/implemented/20210624-flake-rate/flake-rate.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+### Charts
+
+To allow users to see the daily "flakiness" of a test/environment combination, we will have an HTML file at `gs://minikube-flake-rate/flake_chart.html` and a JS file at `gs://minikube-flake-rate/flake_chart.js`. These will fetch `gs://minikube-flake-rate/data.csv` and parse it into Google Charts allowing us to visualize the "flakiness" over time. This can help track down exactly when a test became "flake-y" by telling us the commits associated with each test date. The flake rate charts will use two query parameters (e.g. `google.com?these=are&three=query&param=eters`): test which will control which test to view (`TestFunctional/parallel/LogsCmd`), and env which will control the environment to view (e.g. `Docker_Linux`). By hosting this in a GCS bucket, we can avoid needing to create actual servers to manage this. Since these files are incredibly lightweight, there is little concern over the workload of hosting these files.
```
Comment (sharifelgamal) [inline enhancements/implemented/20210624-flake-rate/flake-rate.md:47]: could we get an example of a link to an actual chart here?

File: enhancements/implemented/20210624-flake-rate/flake-rate.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Your inspiring proposal title
```
[the remark this answers is not in the document]
Comment (andriyDev) [reply enhancements/implemented/20210624-flake-rate/flake-rate.md:1]: Awww that title really spoke to me :'( hahaha, fixed!

File: enhancements/implemented/20210624-flake-rate/flake-rate.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Please review this proposal with the following priorities:
+
+*   Does this fit with minikube's [principles](https://minikube.sigs.k8s.io/docs/concepts/principles/)?
+*   Are there other approaches to consider?
+*   Could the implementation be made simpler?
+*   Are there usability, reliability, or technical debt concerns?
+
+## Summary
+
+As of June 2021, public users have no way to view the flake rates of integration tests. This can make it tricky to determine whether an individual PR is causing a new error, or if the test failure is just a flake, or if the test is entirely broken. While each test failure should be investigated, sometimes an unrelated test fails, and knowing that the test has been "flake-y" can increase confidence in a particular PR.
```
[the remark this answers is not in the document]
Comment (andriyDev) [reply enhancements/implemented/20210624-flake-rate/flake-rate.md:17]: Done.

File: enhancements/implemented/20210624-flake-rate/flake-rate.md (hunk 1 above)
Comment (andriyDev) [reply enhancements/implemented/20210624-flake-rate/flake-rate.md]: Replaced random google link with actual link.

**Review by sharifelgamal — APPROVED** [review]

**Pull request merged**